app/javascript/components/student/editor/CodeEditor.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useState } from 'react'
+
+export function CodeEditor({ onSubmit }: { onSubmit: (code: string) => void }) {
+  const [code, setCode] = useState('')
+
+  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault()
+
+    onSubmit(code)
+  }
+
+  function handleChange(e: React.ChangeEvent<HTMLTextAreaElement>) {
+    setCode(e.target.value)
```

Comment:
Causes a re-render on each key stroke. Use an uncontrolled component with a ref to mitigate this.

Re-renders are not bad, but it's unnecessary and this will probably be slow on low end devices because the code will be big 😁